lib/client/api.go

```diff
@@ -1264,6 +1260,9 @@ type DTAutoEnrollFunc func(context.Context, devicepb.DeviceTrustServiceClient) (
 // TeleportClient is NOT safe for concurrent use.
 type TeleportClient struct {
 	Config
+	exitStatus int
+	statusMu   sync.RWMutex
```

Comment:
I wouldn't use an RWMutex for something so trivial. A regular mutex should be fine here.